pvlib/spectrum/mismatch.py

```diff
@@ -0,0 +1,219 @@
+
+    Returns
+    -------
+    am15g: pandas.Series
+        The AM1.5g standard spectrum indexed by wavelength in nm. [(W/m^2)/nm]
+
+    Notes
+    -----
+    This function uses linear interpolation.  If the values in ``wavelength``
+    are too widely spaced, the integral of the spectrum may deviate from the
+    standard value of 1000.37 W/m^2.
+
+    The values in the data file provided with pvlib-python are copied from an
+    Excel file distributed by NREL, which is found here:
+    https://www.nrel.gov/grid/solar-resource/assets/data/astmg173.xls
+
+    More information about reference spectra is found here:
+    https://www.nrel.gov/grid/solar-resource/spectra-am1.5.html
+
+    References
+    ----------
+    .. [1] ASTM "G173-03 Standard Tables for Reference Solar Spectral
+        Irradiances: Direct Normal and Hemispherical on 37° Tilted Surface."
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Aug. 2022
+
+    pvlib_path = pvlib.__path__[0]
+    filepath = os.path.join(pvlib_path, 'data', 'astm_g173_am15g.csv')
+
+    am15g = pd.read_csv(filepath, index_col=0).squeeze()
+
+    if wavelength is not None:
+        interpolator = interp1d(am15g.index, am15g,
+                                kind='linear',
+                                bounds_error=False,
+                                fill_value=0.0,
+                                copy=False,
+                                assume_sorted=True)
+
+        am15g = pd.Series(data=interpolator(wavelength), index=wavelength)
+
+    am15g.index.name = 'wavelength'
+    am15g.name = 'am15g'
+
+    return am15g
+
+
+def calc_spectral_mismatch(sr, e_sun, e_ref=None):
+    """
+    Calculate spectral mismatch between a test device and broadband reference
+    device under specified solar spectral irradiance conditions.
+
+    Parameters
+    ----------
+    sr: pandas.Series
+        The relative spectral response of one (photovoltaic) test device.
+        The index of the Series must contain wavelength values in nm. [-]
+
+    e_sun: pandas.DataFrame or pandas.Series
+        One or more measured solar irradiance spectra in a pandas.DataFrame
+        having wavelength in nm as column index.  A single spectrum may be
+        be given as a pandas.Series having wavelength in nm as index.
+        [W/m^2/nm]
+
+    e_ref: pandas.Series, optional
+        The reference spectrum to use for the mismatch calculation.
+        The index of the Series must contain wavelength values in nm.
+        The default is the ASTM G173-03 global tilted spectrum. [W/m^2/nm]
+
+    Returns
+    -------
+    smm: pandas.Series or float if a single measured spectrum is provided. [-]
+
+    Notes
+    -----
+    If the default reference spectrum is used it is linearly interpolated
+    to the wavelengths of the measured spectrum.  To achieve alternate
+    behavior e_ref can be transformed before calling this function and
+    provided as an argument.
```

Comment:
```suggestion
    If the default reference spectrum ``e_ref`` is used, it is linearly interpolated
    to the wavelengths of the measured spectrum ``e_sun``. 
```

What is meant by "To achieve alternate behavior e_ref can be transformed before calling this function and provided as an argument." If `e_ref` is default or input, it is interpolated to the frequencies of `sr`, so it's not clear what the alternate behavior is.